### ISSUE TITLE
Add the queryParam() method

### ISSUE THREAD

**Issue by WindBridges** [crwlrsoft/url#27]
It is often necessary to update just one parameter in a query, for example:
```
        $url = new Url($link);
        $params = $url->queryArray();
        $params['foo'] = 'bar';
        $url->queryArray($params);
```

It would be great to do this in one call:
```
        $url = new Url($link);
        $url->queryParam('foo', 'bar');
```

And also the getter:
```
        $url = new Url($link);
        $url->queryParam('foo', 'bar');
```

And also for removal:
```
        $url = new Url($link);
        $url->queryParam('foo', null);
        // or
        $url->removeQueryParam('foo');
```

**Comment by otsch** [comment]
Hey, thanks for that input 🙂👍🏻
You're absolutely right, I also just recently thought about implementing this! 😅
I'll try to find some time for this as soon as possible 👍🏻

**Comment by otsch** [comment]
I just thought: as query strings can also be sent in the request body of a POST request, it probably makes sense to extract this to a separate query-string package. I think this also makes sense from the perspective that there can be multiple levels like: `param[foo][]=val&param[foo][]=val2&param[bar]=value`. So it will become a bit more complex than `$url->queryParam('foo', 'bar')`. 

I'll play around with this and try to find a good solution. In order to add this functionality to the `Url` class, I'm thinking of adding a method `queryString` that'll return an instance of that new separate `QueryString` implementation and thus allowing for easier/flexible query string manipulation.

**Comment by otsch** [comment]
So, I've now added the new `query-string` package and implemented it in the url package in v1.2. As the new package requires PHP 8 it's currently not yet a dependency of url v1.2, so you have to manually require the `crwlr/query-string` package to use it via the url package.

You can find the docs at: 
https://www.crwlr.software/packages/url/v1.2/query-string 
https://www.crwlr.software/packages/query-string/v1.0/getting-started 

Hope you'll like it 🙂
